src/blog/2023/07/how-to-deploy-a-basic-opc-ua-server-in-node-red.md

```diff
@@ -0,0 +1,686 @@
+
+![OPC Information Model](./images/opc-ua-1/opc-information-model.png)
+Devices can be described as objects that give a holistic view of the device, beyond simply the raw value.  To construct a device object, we can take different individual attributes associated with a device, such as the transmitter raw value, transmitter fault flag, alarm setpoint, and combine them, similar to how user-defined datatypes (UDTs) are objects used to represent devices in PLCs.  The information model also defines a folder structure, to allow devices information to reside in a structured hierarchy.  Using the example temperature transmitter above, an example folder structure can be constructed as follows:
+
+`/Root/Objects/Calcinator 1 PLC/Temperature Transmitters/Tank 1 Temperature/Transmitter Value`
+
+This folder structure will be exposed via the OPC Client browser, allowing end-users to easily “drill down” to individual node information in a logical manner.
+
+![OPC Client Browser](./images/opc-ua-1/opc-client-browser.png)
+In summary, OPC-UA represents a trade-off between complex information modeling, with the versatility for that data to be consumed by devices and systems all the way up the automation pyramid layers.  The data does not have to pass through subsequent automation layers on the way up, nor does the data need to undergo any conversion along the way.  
+
+![OPC-UA Distributed Model](./images/opc-ua-1/OPC-UA-distributed-model.jpg)
+The OPC client simply needs to subscribe to the OPC Server endpoint url (ex. opc.tcp://server.address), and the client will be able to browse the structured OPC data as it’s modeled in the server.  Any client will receive the information in the same manner, regardless if it’s a PLC, SCADA, MES, or ERP system.  [This opens the possibility for horizontal and vertical system integration in a standardized manner.](https://ifr.org/post/faster-robot-communication-through-the-opc-robotics-companion-specification)  Additionally, the more information that is exposed about a device, the easier it is to track, and use said data to autonomously reconfigure, or pre-emptively take maintenance actions.  
+
+## Deploying an Example OPC-UA Server in Node-RED
+
+With some background on OPC-UA and how information is modeled in mind, we can take a look at the [node-red-contrib-opcua-server](https://flows.nodered.org/node/node-red-contrib-opcua-server) node, which is merely a compact version of the [node-red-contrib-opcua](https://flows.nodered.org/node/node-red-contrib-opcua) node that only focuses on the OPC-UA server and hence requires less dependencies, maintained by [Klaus Landsdorf](https://github.com/biancode).  
```

Comment:
I would delete 'maintained by Klaus'. It is kind out of context for this sentence. If you think it is important, please start a new sentence.

Reply:
it's not necessary.  I will remove it.